src/Payum/Be2Bill/Api.php

```diff
@@ -138,6 +144,7 @@ public function __construct(array $options, HttpClientInterface $client = null)
 
         $this->options = $options;
         $this->client = $client ?: HttpClientFactory::create();
+        $this->messageFactory = $options['httplug.message_factory'] ?: MessageFactoryDiscovery::find();;
```

Comment:
I'd prefer to have it injected as third argument (optional) and if not passed use a discovery logic. FYI I am going to make this argument not optional in 2.x version as they managed by gateway factory.